tests/api/api/test_tags.py

```diff
@@ -0,0 +1,549 @@
+# Copyright 2018 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import http
+import json
+import uuid
+
+import fastapi.testclient
+import sqlalchemy.orm
+
+import mlrun.api.schemas
+
+API_PROJECTS_PATH = "projects"
+API_ARTIFACTS_PATH = "projects/{project}/artifacts"
+API_ARTIFACTS_PATH_WITH_TAG = API_ARTIFACTS_PATH + "?tag={tag}"
+API_TAGS_PATH = "projects/{project}/tags/{tag}"
+STORE_API_ARTIFACTS_PATH = API_ARTIFACTS_PATH + "/" + "{uid}/{key}?tag={tag}"
+
+
+class TestArtifactTags:
+    project = "test-project"
+
+    def test_overwrite_artifact_tags_by_name_identifier(
+        self, db: sqlalchemy.orm.Session, client: fastapi.testclient.TestClient
+    ):
+        self._create_project(client)
+
+        tag = "tag1"
+        overwrite_tag = "tag2"
+        artifact1_labels = {"artifact_name": "artifact1"}
+        _, _, artifact1_name, artifact1_uid, artifact1_key, _, _ = self._store_artifact(
+            client, tag=tag, labels=artifact1_labels
+        )
+        _, _, artifact2_name, artifact2_uid, artifact2_key, _, _ = self._store_artifact(
+            client, tag=tag
+        )
+
+        response = client.post(
+            API_TAGS_PATH.format(project=self.project, tag=overwrite_tag),
+            json={
+                "kind": "artifact",
+                "identifiers": [
+                    {
+                        "name": artifact1_key,
+                    },
+                ],
+            },
```

Comment:
This block of code is repeated multiple times,  why not put it in a function - `generate_tag_identifiers_json(identifiers)` that will create it? Just to make this code shorter...